modules/core/legacy/v100/genesis.go

```diff
@@ -0,0 +1,40 @@
+package v100
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+
+	clientv100 "github.com/cosmos/ibc-go/modules/core/02-client/legacy/v100"
+	host "github.com/cosmos/ibc-go/modules/core/24-host"
+	"github.com/cosmos/ibc-go/modules/core/types"
+)
+
+// MigrateGenesis accepts exported v1.0.0 IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v1 to v2)
+// - Remove all solo machine consensus states
+// - Remove all expired tendermint consensus states
+func MigrateGenesis(appState genutiltypes.AppMap, clientCtx client.Context, genesisBlockTime time.Time) (genutiltypes.AppMap, error) {
```

Comment:
this needs to be tested. I think I need to add the unpack interfaces functions for solo machine. Will work on this tomorrow